### ISSUE TITLE
Add Support for Agent-to-Agent Communication

### ISSUE THREAD

**Issue by ravediamond** [langchain-ai/agent-protocol#5]
## Problem

The current Agent Protocol specification focuses solely on human-agent interactions through threads and runs. But I think many applications will require many specialized agents working together. Currently, there's no specified way for:

- Agents to discover other agents and their capabilities
- Agents to communicate directly with each other
- Coordinating work between multiple agents
- Delegating tasks between agents
- Sharing context and knowledge between agents

## Proposed Solution
Add new endpoints that enable agent discovery and communication:

```
# Agent Registry
GET /agents
Response:
{
  "agents": [{
    "id": "agent-123",
    "name": "Research Assistant",
    "capabilities": {
      "tasks": ["research", "summarize"],
      "input_types": ["text", "pdf"],
      "output_types": ["text", "json"]
    },
    "status": "active"
  }]
}

# Agent Communication
POST /threads/{thread_id}/messages
{
  "from_agent": "agent_id",
  "to_agent": "agent_id",
  "content": {},
  "type": "request|response"
}

# Agent Task Delegation
POST /threads/{thread_id}/delegate
{
  "delegate_to": "agent_id",
  "permissions": ["read", "write", "execute"],
  "task": {
    "type": "task_type",
    "input": {},
    "constraints": {}
  }
}

# Agent Shared State Creation 
POST /threads/{thread_id}/shared_state
{
  "namespace": "shared_namespace",
  "access": ["agent_id1", "agent_id2"],
  "state": {}
}


```

This solution would provide:
1. A registry where agents can publish their capabilities and discover others
2. Standard message formats for agent-to-agent communication. I think this would need a specific format optimized for agent-to-agent communication.
3. Integration with existing thread model for conversation history

What do you think ?

**Comment by markokraemer** [comment]
I dont think this should be part of it yet! Most multi agent systems dont work reliabely in prod. The Agent To Agent comm can be just a tool. And often is just abstracted in SDK´s although its just a tool call under the hood.

So a more general, better approach would be having a call tool as api  Tool API schema defined here.

**Comment by nfcampos** [comment]
We've taken the first steps towards a registry with the first introspection endpoints in #23. More to come

**Comment by agarciadom** [comment]
I'm part of a research project that is starting to explore this idea of agent-to-agent communications as well, and we are considering both creating a custom A2A protocol, or extending AP.

Given the AP already accommodates for a "custom" streaming mode where we can provide arbitrary objects to be streamed, at the time being we think this may be more a matter of specifying what those objects could be into some type of A2A-specific streaming mode (e.g. where the objects represent proposed alternative solutions for a given task, evaluating a solution proposed by another agent, or providing interim updates on some complicated task that may have its own internal handoffs).

So rather than needing new endpoints, perhaps it could be a new streaming mode. We may play around with this idea and report back :-).
